r/tests/testthat/test-parquet.R

```diff
@@ -234,3 +234,17 @@ test_that("ParquetFileReader $ReadRowGroup(s) methods", {
   expect_true(reader$ReadRowGroups(c(0, 1), 0) == Table$create(x = 1:20))
   expect_error(reader$ReadRowGroups(c(0, 1), 1))
 })
+
+test_that("Error messages are shown when the compression algorithm lz4/snappy
+          is not found", {
+  skip_on_cran()
+  if (codec_is_available("snappy")) {
+    d <- read_parquet(system.file("extdata", "pets.parquet", package = "arrow"))
```

Comment:
This file isn't necessary because the example parquet file in `inst/` is snappy compressed, so you can use that. (Look for `skip_if_not_available("snappy")` throughout this file.)

Reply:
thanks, I'll push when this passed the local test

Reply:
regarding the skip_if_not_available("snappy"), wouldn't we prefer to test and check that the error message is exactly what we want?

Reply:
I ask because codec_is_available("snappy") leaves space for expect_error

Reply:
so far I creates this error (the algorithm name changes):
 Error: Unsupported compressed format 'lz4'. We suggest either setting the right environment variable DARROW_WITH_LZ4=ON to install binaries or setting LIBARROW_MINIMAL=false and then reinstall the package.

Reply:
My point about `skip_if_not_available("snappy")` is that you can find other tests in this file that use a test file with snappy compression already. I wasn't suggesting that you use it in your test.